packages/server/src/api/controllers/row/internal.js

```diff
@@ -375,14 +375,28 @@ exports.exportRows = async ctx => {
   const table = await db.get(ctx.params.tableId)
   const rowIds = ctx.request.body.rows
   let format = ctx.query.format
+  const { columns } = ctx.request.body
   let response = (
     await db.allDocs({
       include_docs: true,
       keys: rowIds,
     })
   ).rows.map(row => row.doc)
 
-  let rows = await outputProcessing(table, response)
+  let result = await outputProcessing(table, response)
+  let rows = []
+
+  // Filter data to only specified columns if required
+  if (columns && columns.length) {
```

Comment:
We should cover this with a unit test - it's quite a fundamental path in the code.

Reply:
This was just the exact same logic that I already added to the external controller and has been deployed for quite a while - I just didn't add it to the internal controller as well. I don't think any of the export stuff is currently covered by unit tests, but I guess I'll write some. I agree though it should have had tests written in the first place.